### ISSUE TITLE
AzCopy don't respect AZCOPY_DEFAULT_SERVICE_API_VERSION when downloading a public blob

### ISSUE THREAD

**Issue by msftcoderdjw** [Azure/azure-storage-azcopy#1328]
### Which version of the AzCopy was used? 
azcopy_windows_amd64_10.8.0
### Which platform are you using? (ex: Windows, Mac, Linux)
Windows, Linux

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.
set AZCOPY_DEFAULT_SERVICE_API_VERSION=2017-11-09
.\azcopy.exe cp "https://\<accountName\>.blob.\<AzureStackStorageEndpoint\>/\<containerName\>/\<blobName\>" \<localPath\>

### What problem was encountered?
"https://\<accountName\>.blob.\<AzureStackStorageEndpoint\>/\<containerName\>/\<blobName\>" is a public blob, and could be downloaded in browser. But when invoking above command: .\azcopy.exe cp "https://\<accountName\>.blob.\<AzureStackStorageEndpoint\>/\<containerName\>/\<blobName\>" \<localPath\> , it fails with "failed to perform copy command due to error: Login Credentials missing. No SAS token or OAuth token is present and the resource is not public"

By capturing the network, I could see AzCopy issued a GET container properties request and the x-ms-version is always 2019-12-12 and doesn't respect the AZCOPY_DEFAULT_SERVICE_API_VERSION environment variable. I was doing this operation against Azure Stack, and 2019-12-12 is not supported in Azure Stack Hub now.

### How can we reproduce the problem in the simplest way?
set AZCOPY_DEFAULT_SERVICE_API_VERSION=2017-11-09
.\azcopy.exe cp "https://\<accountName\>.blob.\<AzureStackStorageEndpoint\>/\<containerName\>/\<blobName\>" \<localPath\>
And monitor the network traffic.

### Have you found a mitigation/solution?
Needs to put a sasToken, e.g.
set AZCOPY_DEFAULT_SERVICE_API_VERSION=2017-11-09
.\azcopy.exe cp "https://\<accountName\>.blob.\<AzureStackStorageEndpoint\>/\<containerName\>/\<blobName\>**\<sasToken\>**" \<localPath\>

**Comment by mohsha-msft** [comment]
This is a bug in SDK and not AzCopy.

```
const (
	// ServiceVersion specifies the version of the operations used in this package.
	ServiceVersion = "2020-04-08"
)
```
We use const ServiceVersion when we construct request in SDK 
`req.Header.Set("x-ms-version", ServiceVersion)`

**Comment by mohsha-msft** [comment]
Fixed in AzCopy Release `10.14.0`
